### DISCUSSION TITLE
x-data assignment, is object or function?

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#4129] [4. General]
Hi,

I have a general question, somewhere I saw example x-data = {}, somewhere I saw example x-data=variable, and variable = (x) => ({}) defined in script section, and I am not sure when I should use {}, when I should use function, or I can just use function always.

Thanks for help me clarify this.
Thanks
Eric Xin

**Comment by ekwoka** [comment]
If you are passing it to Alpine.data it must be a function that returns an object

When writing it in x-data, it needs to be an expression that evaluates to an object or to a function that returns an object.

All Alpine expressions are encapsulated into functions already.

**Reply by ericxin1982** [reply]
Thanks so much!
